pymeasure/instruments/teledyne/teledyneMAUI.py

```diff
@@ -0,0 +1,211 @@
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.teledyne.teledyne_oscilloscope import TeledyneOscilloscope, \
+    TeledyneOscilloscopeChannel, _results_list_to_dict
+
+
+class TeledyneMAUIChannel(TeledyneOscilloscopeChannel):
+    """Base class for channels on a :class:`TeledyneMAUI` device."""
+
+    # For some reason "20MHZ" is registered as "ON"
+    BANDWIDTH_LIMITS = ["OFF", "ON", "200MHZ"]
+
+    TRIGGER_SLOPES = {"negative": "NEG", "positive": "POS"}
+
+    # Reset listed values for existing commands:
+    bwlimit_values = BANDWIDTH_LIMITS
+
+    def autoscale(self):
+        """Perform auto-setup command for channel."""
+        self.write("AUTO_SETUP FIND")
+
+    # noinspection PyIncorrectDocstring
+    def setup(self, **kwargs):
+        """Setup channel. Unspecified settings are not modified.
+
+        Modifying values such as probe attenuation will modify offset, range, etc. Refer to
+        oscilloscope documentation and make multiple consecutive calls to setup() if needed.
+        See property descriptions for more information.
+
+        :param bwlimit:
+        :param coupling:
+        :param display:
+        :param offset:
+        :param probe_attenuation:
+        :param scale:
+        :param trigger_coupling:
+        :param trigger_level:
+        :param trigger_slope:
+        """
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+    @property
+    def current_configuration(self):
+        """Get channel configuration as a dict containing the following keys:
+
+        - "channel": channel number (int)
+        - "attenuation": probe attenuation (float)
+        - "bandwidth_limit": bandwidth limiting, parsed for this channel (str)
+        - "coupling": "ac 1M", "dc 1M", "ground" coupling (str)
+        - "offset": vertical offset (float)
+        - "display": currently displayed (bool)
+        - "volts_div": vertical divisions (float)
+        - "trigger_coupling": trigger coupling can be "dc" "ac" "highpass" "lowpass" (str)
+        - "trigger_level": trigger level (float)
+        - "trigger_slope": trigger slope can be "negative" "positive" "window" (str)
+        """
+        ch_setup = {
+            "channel": self.id,
+            "attenuation": self.probe_attenuation,
+            "bandwidth_limit": self.bwlimit[f"C{self.id}"],
+            "coupling": self.coupling,
+            "offset": self.offset,
+            "display": self.display,
+            "volts_div": self.scale,
+            "trigger_coupling": self.trigger_coupling,
+            "trigger_level": self.trigger_level,
+            "trigger_slope": self.trigger_slope
+        }
+        return ch_setup
+
+
+class TeledyneMAUI(TeledyneOscilloscope):
+    """A base class for the MAUI-type of Teledyne oscilloscopes.
+
+    This class is not exactly device specific. Nonetheless, it might already work out of the box and
+    therefore this class not abstract.
```

Comment:
I would state it differently:
```suggestion
    This base class works out of the box. Some properties, especially the number of channels,
    might have to be adjusted to the actual device.
```